lib/worker/messaging.js

```diff
@@ -0,0 +1,120 @@
+const { AsyncLocalStorage } = require('async_hooks');
+function safeStringify(data) {
+  try {
+    const seen = new WeakSet();
+
+    return JSON.stringify(data, (_key, value) => {
+      if (typeof value !== "object" || value === null) {
+        return value;
+      }
+
+      if (seen.has(value)) {
+        return '[Circular]';
+      }
+
+      seen.add(value);
+      return value;
+    });
+  } catch (error) {
+    return JSON.stringify({
+      message: error.message,
+      stack: error.stack,
+    });
+  }
+}
+
+function stringify(input) {
+  if (typeof input === 'string') {
+    const start = input[0];
+    const end = input[input.length - 1];
+
+    if (
+      (start === '[' && end === ']') ||
+      (start === '{' && end === '}') ||
+      (start === '"' && end === '"')
+    ) {
+      return input;
+    }
+  }
+
+  return safeStringify(input);
+}
+
+/**
+ * @typedef {Object} Store
+ * @property {string} [key]
+ * @property {string} [type]
+ * @property {string} [filePath]
+ * @property {string} [projectPath]
+ */
+/** @type {AsyncLocalStorage<Store|void>} */
+const store = new AsyncLocalStorage();
+
+function runWith (data, handler) {
+  store.run(
+    {
+      key: data?.key,
+      type: data?.type,
+      filePath: data?.filePath,
+      projectPath: data?.projectPath,
+    },
+    handler,
+  );
+}
```

Comment:
This allows any messages or errors thrown to have the requests context in the response.
This is useful when its an uncaught exception.